### PR TITLE
Test java plugin parallel configuration

### PR DESCRIPTION
**Pull request by 6hundreds**

### CODE REVIEW

File: subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/isolated/IsolatedProjectsJavaPluginIntegrationTest.groovy [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+        then:
+        fixture.assertStateStored {
+            projectsConfigured(":buildSrc", ":", ":a", ":b")
+            buildModelCreated()
+            modelsCreated(":a", ":b")
+        }
+    }
+
+    def "java projects with a valid cross-references can be configured in a parallel"() {
```
Comment (bamboo) [inline subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/isolated/IsolatedProjectsJavaPluginIntegrationTest.groovy:49]: 🧪 Shouldn't this test include a _valid_ cycle between `a` and `b`?

Comment (bamboo) [reply subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/isolated/IsolatedProjectsJavaPluginIntegrationTest.groovy:49]: I see the cycle now. It's between the `implementation project(':b')` and `implementation(project(":a")) { ... }`

**Review by bamboo — APPROVED** [review]
This is still not exercising the cycle enough to expose deadlocks and other potential issues but it's a good start. Let's learn more with the gradle-profiler bencharmk and improve the test as needed.

**Comment by 6hundreds** [comment]
@bot-gradle test and merge

**Pull request merged**